src/components/FrontPage.js

```diff
@@ -0,0 +1,34 @@
+import styled from "styled-components";
+import NavBar from "./NavBar";
+import {BrowserRouter, Route, Routes} from "react-router-dom";
+import FileUpload from "./upload/FileUpload";
+import EditorPage from "./editor/EditorPage";
+import GeneratorPage from "./generator/GeneratorPage";
+
+
+const FrontPageRoot = styled.div`
+  height: 100vh;
+  width: 100%;
+  min-width: 800px;
+  display: grid;
+  grid-template-columns: 200px auto;
+  padding: 10px;
+`;
+
+const FrontPage = () => {
+
+    return (
+        <FrontPageRoot>
+            <BrowserRouter>
+                <NavBar/>
+                <Routes>
+                    <Route path="/create" element={<FileUpload/>}/>
+                    <Route path="/edit" element={<EditorPage/>}/>
+                    <Route path="/generate" element={<GeneratorPage/>}/>
+                </Routes>
+            </BrowserRouter>
+        </FrontPageRoot>
+    );
+};
+
+export default FrontPage;
```

Comment:
Missing newlines in some files. Found out recently that you can configure all JetBrains IDEs to make that automatically:
`Settings -> Editor -> General -> Ensure line feed at file end on save`